src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -62,16 +62,22 @@ final class CachedAppConfiguration {
 					guard let config = self?.store.appConfig else {
 						fatalError("App configuration cache broken!") // in `where` we trust
 					}
-					completion?(.success(config))
+					self?.completeOnMain(completion: completion, result: .success(config))
 
 					// keep track of last successful fetch
 					self?.store.lastAppConfigFetch = Date()
 				default:
-					completion?(.failure(error))
+					self?.completeOnMain(completion: completion, result: .failure(error))
 				}
 			}
 		}
 	}
+
```

Comment:
Quick explanation why you did this:

```suggestion

	// Prevents failing main thread checks because UI is accessing the result directly.
```